lib/sidebar_pages.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: false
+
+require 'forwardable'
+
+module SwedbankPay
+  # Provides filtering and utility methods on the pages used to build the Sidebar
+  class SidebarPages
+    include Enumerable
+    extend Forwardable
+    def_delegators :@pages, :each, :<<
+
+    def initialize(pages)
+      raise ArgumentError, 'Pages must be a Hash' unless pages.is_a? Hash
+
+      @pages = tree(pages)
+    end
+
+    def to_s
+      string(false)
+    end
+
+    def inspect
+      string(true)
+    end
+
+    def count
+      count_pages(@pages)
+    end
+
+    private
+
+    def string(inspect)
+      s = "<#{self.class} ##{count}"
+
+      if inspect
+        s << ":\n"
+        @pages.each do |page|
+          s << "#{page}\n"
+        end
+      end
+
+      s << '>'
+      s
+    end
+
+    def count_pages(pages)
+      return 0 if pages.nil? || pages.empty?
+
+      count = pages.length
+
+      pages.each do |page|
+        count += count_pages(page.children)
+      end
+
+      count
+    end
+
+    def tree(pages)
```

Comment:
Codacy found an issue: [Assignment Branch Condition size for tree is too high. [<5, 18, 4> 19.1/15]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)